src/SDK/Common/Dsn/Factory.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Common\Dsn;
+
+use InvalidArgumentException;
+
+final class Factory implements FactoryInterface
+{
+    /** @var array  */
+    public const REQUIRED_ATTRIBUTES = [
+        DsnInterface::TYPE_ATTRIBUTE,
+        DsnInterface::SCHEME_ATTRIBUTE,
+        DsnInterface::HOST_ATTRIBUTE,
+    ];
+
+    private array $defaults;
+
+    public function __construct(array $defaults = [])
+    {
+        $this->defaults = $defaults;
+    }
+
+    public static function create(array $defaults = []): self
+    {
+        return new self($defaults);
+    }
+
+    /**
+     * @phan-suppress PhanUndeclaredClassAttribute
+     */
+    #[\ReturnTypeWillChange]
```

Comment:
will it change? I usually see this used to quieten 8.1 deprecations warnings, but this looks like it's returning the type is should

Reply:
Yes, the interface defines a return type of `DsnInterface`